windows/datetimepicker.cpp

```diff
@@ -131,6 +134,72 @@ static void uiDateTimePickerMinimumSize(uiWindowsControl *c, int *width, int *he
 	*height = y;
 }
 
+static BOOL onWM_NOTIFY(uiControl *c, HWND hwnd, NMHDR *nmhdr, LRESULT *lResult)
+{
+	uiDateTimePicker *d = uiDateTimePicker(c);
+
+	if (nmhdr->code != DTN_DATETIMECHANGE)
+		return FALSE;
+	(*(d->onChanged))(d, d->onChangedData);
+	*lResult = 0;
+	return TRUE;
+}
+
+static void fromSystemTime(LPSYSTEMTIME systime, struct tm *time)
+{
+	memset(time, 0, sizeof(struct tm));
+	time->tm_sec = systime->wSecond;
+	time->tm_min = systime->wMinute;
+	time->tm_hour = systime->wHour;
+	time->tm_mday = systime->wDay;
+	time->tm_mon = systime->wMonth - 1;
+	time->tm_year = systime->wYear - 1900;
+	time->tm_wday = systime->wDayOfWeek;
+	time->tm_isdst = -1;
+}
+
+static void toSystemTime(const struct tm *time, LPSYSTEMTIME systime)
+{
+	memset(systime, 0, sizeof(SYSTEMTIME));
+	systime->wYear = time->tm_year + 1900;
+	systime->wMonth = time->tm_mon + 1;
+	systime->wDayOfWeek = time->tm_wday;
+	systime->wDay = time->tm_mday;
+	systime->wHour = time->tm_hour;
+	systime->wMinute = time->tm_min;
+	systime->wSecond = time->tm_sec;
+}
+
+static void defaultOnChanged(uiDateTimePicker *d, void *data)
+{
+	// do nothing
+}
+
+void uiDateTimePickerTime(uiDateTimePicker *d, struct tm *time)
+{
+	SYSTEMTIME systime;
+
+	if (DateTime_GetSystemtime(d->hwnd, &systime) != GDT_VALID)
```

Comment:
For consistency, use the `SendMessageW()` messages directly, instead of the wrapper macros.